### PR TITLE
[PKG-2498] dbt-extractor 0.4.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/dbt-labs/dbt-extractor/blob/main/CHANGELOG.md
License: https://github.com/dbt-labs/dbt-extractor/blob/main/LICENSE
Requirements: https://github.com/dbt-labs/dbt-extractor/blob/main/pyproject.toml

Actions:
1. Reset build number to `0`
2. Add `--no-build-isolation` to `script`
3. Remove cross-compilation stuff from `build`
4. Add `doc_url`
5. Add `license_family`
6. Add `description`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,16 +10,16 @@ source:
...
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}
-    - {{ compiler('rust') }}
+    - {{ compiler('rust') }}  # [not win]
+    - rust_win-64             # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
This is added by compiler('rust')

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,16 +10,16 @@ source:
...
+  skip: true  # [py<36 or s390x]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}
-    - {{ compiler('rust') }}
+    - {{ compiler('rust') }}  # [not win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - {{ compiler('rust') }}
```

In conda_build_config.yaml, two rust compilers are configured, rust and rust-gnu, resulting in two variants.
It looks like you had an issue with rust-gnu (possibly from having compiler('c') set).
If you only want to build the rust variant, it is better to skip rust-gnu with a selector.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,16 +10,16 @@ source:
...
+  number: 0
+  # maturin isn't availabe on s390x
+  skip: true  # [py<36 or s390x]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
The C compiler is a run dependency of rust_win-64. (respectively, rust-gnu_win-64 depends on m2w64-toolchain)

File: recipe/meta.yaml [hunk 4]
```diff
@@ -11,16 +10,16 @@ source:
   sha256: 75b1c665699ec0f1ffce1ba3d776f7dfce802156f22e70a7b9c8f0b4d7e80f42
 
 build:
-  skip: true  # [py<36]
-  number: 1
-  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  # maturin isn't availabe on s390x
+  skip: true  # [py<36 or s390x]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [py<36 or s390x or (win and rust_compiler == "rust-gnu")]
```

**Review by cbouss — COMMENTED** [review]
I think it would be better to skip rust-gnu and keep the rust compiler setting simpler.

**Review by boldorider4 — APPROVED** [review]
@cbouss points need to be addressed, otherwise LGTM.

**Comment by skupr-anaconda** [comment]
@cbouss After removing the C compiler we have this error on `linux`:
`["  ERROR (dbt-extractor,lib/python3.10/site-packages/dbt_extractor/dbt_extractor.abi3.so): .. but ['libgcc-ng'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]`
Looks like it can be related to this `rust_activation-feedstock` script https://github.com/AnacondaRecipes/rust-activation-feedstock/blob/5328d1600458dd2edb430529ae6128a363dca9cb/recipe/activate.sh#L50-L51

It seems like the C compiler is still required, at least, for `linux`

**Review by cbouss — APPROVED** [review]

**Pull request merged**